typescript/src/hex.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * Represents a hexadecimal value.
+ */
+export class Hex {
+  protected readonly _hex: Buffer
+
+  constructor(value: Buffer | string) {
```

Comment:
Looks like we use the `from` factory method to create new `Hex` instances across the codebase. Maybe we can make the constructor `private` to force a uniform way of creating `Hex` instances for the future?

Reply:
a987ecfe39ab79e9168a6f330c4f142a48f59243